### PR TITLE
Microsoft Azure extension pack

### PR DESCRIPTION
**Pull request by kaizimmerm**
A set of extensions for optimized integration of hawkBit update server into Microsoft Azure.

In addition it includes a reference runtime, i.e. hawkbit-update-server-azure, that packages all the extensions together in a ready to go deployment.

The runtime includes:

- hawkBit DDI, management API and management UI
- [Azure SQL Database](https://azure.microsoft.com/en-us/services/sql-database/) for the metadata repository
- [Azure Storage](https://azure.microsoft.com/en-us/services/storage/) for software artifact binary persistence

This is a start. However, further topics come to mind:
 
* hawkBit inner cluster communication based on [Azure EventHub's new Kafka support](https://azure.microsoft.com/en-us/blog/azure-event-hubs-for-kafka-ecosystems-in-public-preview/) (however, that needs an update of Spring Cloud/Boot as this feature requests Kafka 1.0 support),
* ARM template for simplified deployment.
* Azure AD integration for Management UI and API
* Azure Kubernetes Service (AKS) deployment for improved scheduling

### CODE REVIEW

File: hawkbit-azure-extensions-parent/pom.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<!--
```
Comment (schabdo) [inline hawkbit-azure-extensions-parent/pom.xml]: I'd prefer the root named e.g. `hawkbit-extension-azure` or `hawkbit-extensions-azure` so it is more consistent with the other extensions

File: hawkbit-extended-runtimes/hawkbit-update-server-azure/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+EXPOSE 8080
+
+VOLUME "$HAWKBIT_HOME/data"
+
+WORKDIR $HAWKBIT_HOME
+ENTRYPOINT ["java","-jar","hawkbit-update-server.jar","-Xmx768m -Xmx768m -XX:MaxMetaspaceSize=250m -XX:MetaspaceSize=250m -Xss300K -XX:+UseG1GC -XX:+UseStringDeduplication -XX:+UseCompressedOops -XX:+HeapDumpOnOutOfMemoryError"]
+
+# Add the service itself
+ARG JAR_FILE
```
Comment (schabdo) [inline hawkbit-extended-runtimes/hawkbit-update-server-azure/Dockerfile]: Can this be removed? It's never been used within this file

File: hawkbit-extensions-azure/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# hawkBit extensions for Microsoft Azure
+
+This module combines a set of extensions for optimized integration of hawkBit update server into Microsoft Azure.
+
+In addition we provide a pre-packaged [runtime including the Azure extensions](/hawkbit-runtimes-parent/hawkbit-update-server-azure).
```
Comment (schabdo) [inline hawkbit-extensions-azure/README.md]: Link is pointing to old file structure

File: hawkbit-extended-runtimes/hawkbit-update-server-azure/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+ADD target/hawkbit-update-server-azure-0.3.0-SNAPSHOT.jar hawkbit-update-server.jar
```
Comment (schabdo) [inline hawkbit-extended-runtimes/hawkbit-update-server-azure/Dockerfile]: I won't hardcode the version here. Ah, I see that was the intention for `JAR_FILE` 👍

Comment (kaizimmerm) [reply hawkbit-extended-runtimes/hawkbit-update-server-azure/Dockerfile]: I agree, we should find a better solution for that. However, right now its for manual build only anyhow.

Comment (schabdo) [reply hawkbit-extended-runtimes/hawkbit-update-server-azure/Dockerfile]: Well, maybe you could that a default for the `ARG` and use it within the path. So you could pass a different value at build time

**Comment by schabdo** [comment]
Awesome, great job @kaizimmerm. Looking forward to see HawkBit running on Azure!

**Pull request merged**